### DISCUSSION TITLE
It is ok to useState in client middleware?

### DISCUSSION THREAD

**Discussion by ginrei** [nuxt/nuxt#17495] [Questions] [answered]
[The guide page](https://v3.nuxtjs.org/guide/features/data-fetching) says
>  useFetch, useLazyFetch, useAsyncData and useLazyAsyncData only work during setup or Lifecycle Hooks

[It ](https://v3.nuxtjs.org/guide/features/state-management)also says
> useState only works during setup or Lifecycle Hooks .

I am a bit confused about the range of the term "**_during setup_**".
Is it ok to use useState in client middleware?
Client middleware, which is, of-course be  coded among  **_setup_** section of components or pages.

If [useFetch should be avoided in the middleware](https://github.com/nuxt/framework/discussions/3284#discussion-3883090), so is useState I guess.
Wrapping useState in composable may not  change the _out-of-setup_ circumstance.

```
type User = {
  auth: boolean
  username: string
}

export default defineNuxtRouteMiddleware(async (to, from) => {
  const user = useState<User>('user') // How about this?
  
  if (user.value === undefined) {
    const { data: user } = await useFetch( // I know this should be replaced with $fetch
      `/api/user`
    )
    useState('user', () => user)
  }
  
  if (user.value.auth !== true) {
    return abortNavigation()
  }
  return
})
```

**Comment by wobsoriano** [comment, marked as answer]
That's perfectly fine. It is not fetching any data like what `useFetch` does.

Take a look at this example https://supabase.nuxtjs.org/advanced

that `useSupabaseUser` composable wraps a `useState`
